### PR TITLE
Feature/wf 1143 add ng api mock

### PR DESCRIPTION
**Pull request by ankypant**
This PR is for adding ng-apimock to GSA.
I have mocked all the calls that are made by GSA at the moment.
README contains further instruction on how to run the mocks

### CODE REVIEW

File: apps/golden-sample-app/src/app/app.module.ts [hunk 1]
```diff
@@ -97,32 +97,34 @@ import { ActivityMonitorModule } from './auth/activity-monitor';
...
-            if (type === 'token_received' || type === 'token_refreshed') {
-              // Set the cookie on the app domain
-              cookieService.set(
-                'Authorization',
-                `Bearer ${oAuthService.getAccessToken()}`
-              );
-            }
-          });
+    environment.mockEnabled
+      ? []
```
Comment (manisha-backbase) [inline apps/golden-sample-app/src/app/app.module.ts:101]: should it be an empty object?

Comment (ankypant) [reply apps/golden-sample-app/src/app/app.module.ts:101]: Provider accepts 

`export declare type Provider = TypeProvider | ValueProvider | ClassProvider | ConstructorProvider | ExistingProvider | FactoryProvider | any[];`

If you wish to add any other type apart from a provider then it should be an array

File: README.md [hunk 2]
```diff
@@ -121,6 +121,10 @@ Run `npm run start` for a dev server. Navigate to <http://0.0.0.0:4200/>. The ap
 
 Credentials to login can be found [here](https://backbase.io/developers/documentation/api-sandbox/runtime_data/user_credentials/).
 
+## Running the app with Mocks
+
+Run `npm run mock-server` for bringing up the mock server followed by `npm run start:mocks` for running the application locally with mocks.
```
[the remark this answers is not in the document]
Comment (manisha-backbase) [reply README.md:126]: Also, when somebody adds a new feature to GSA, should we make it mandatory to add the mocks for that feature. Else the page will break.

[the remark this answers is not in the document]
Comment (manisha-backbase) [reply README.md:126]: Maybe we can create a separate ticket for these two tasks.

**Review by manisha-backbase — DISMISSED** [review]

**Review by manisha-backbase — DISMISSED** [review]

**Review by manisha-backbase — APPROVED** [review]

**Pull request merged**